File: Python/BOJ/15650.py
import sys
readl = sys.stdin.readline

def solution(current, num):
  if len(current) == m:
    for num in current:
      print(num, end = " ")
    print("")
    return

  for i in range(num + 1, n + 1):
    if not i in current:
      solution(current + [i], i)
    
n, m = map(int, readl().split())

solution([], 0)

File: Swift/BOJ/15000/15650.swift
//
//  15650.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/04/02.
//

import Foundation

func dfs(_ current: [Int], _ num: Int) {
    if current.count == m {
        print(current.map { "\($0)" }.joined(separator: " "))
        return
    }
    
    for i in num...n {
        if current.contains(i) {
            continue
        }
        
        dfs(current + [i], i)
    }
}

let nm = readLine()!.split { $0 == " " }.map { Int(String($0))! }
let (n, m) = (nm[0], nm[1])

dfs([], 1)
